### PR TITLE
 [18/N] Fix clang-tidy warnings in jit 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #132753

### CODE REVIEW

File: torch/csrc/jit/mobile/debug_info.cpp [hunk 1]
```diff
@@ -140,7 +139,7 @@ MobileDebugTable::MobileDebugTable(
       }
 
       for (auto& val : lines.toTuple()->elements()) {
-        auto tup_elems = std::move(*std::move(val).toTuple()).elements();
```
Comment (Skylion007) [inline torch/csrc/jit/mobile/debug_info.cpp:143]: This is valid though? Is it used somewhere else in the loop? This calls the rvalue overload of toTuple if there are multiple implementations.

File: torch/csrc/jit/mobile/debug_info.cpp [hunk 2]
```diff
@@ -159,7 +158,7 @@ MobileDebugTable::MobileDebugTable(
         reader->getRecord(callstack_debug_file);
     CallStackDebugInfoUnpickler unpickler;
     callstack_ptr_map_ = unpickler.unpickle(
-        std::move(callstack_data), callstack_data_size, source_range_map, cu);
+        callstack_data, callstack_data_size, source_range_map, cu);
```
Comment (Skylion007) [inline torch/csrc/jit/mobile/debug_info.cpp:161]: Unpickler doesn't take rvalue?

File: torch/csrc/jit/mobile/import.cpp [hunk 3]
```diff
@@ -146,7 +145,7 @@ c10::intrusive_ptr<c10::ivalue::Object> objLoaderMobile(
     custom_class_type->getMethod("__setstate__").run(stack);
     return obj;
   } else {
-    auto dict = std::move(input).toGenericDict();
+    auto dict = input.toGenericDict();
```
Comment (Skylion007) [inline torch/csrc/jit/mobile/import.cpp:148]: Likewise what was wrong before here?

File: torch/csrc/jit/mobile/promoted_prim_ops.cpp [hunk 4]
```diff
@@ -94,7 +94,7 @@ void device(Stack& stack) {
 
 void device_with_index(Stack& stack) {
   std::string type = pop(stack).toStringRef();
-  int index = pop(stack).toInt();
+  auto index = pop(stack).toInt();
   std::string device_str = type + ":" + std::to_string(index);
```
Comment (Skylion007) [inline torch/csrc/jit/mobile/promoted_prim_ops.cpp:98]: Not should use libfmt or perhaps char overload. Might actually be a bit cleaner with append here.

File: torch/csrc/jit/mobile/flatbuffer_loader.cpp [hunk 5]
```diff
@@ -93,7 +92,7 @@ class FlatbufferLoader final {
 
   typedef TypePtr (*TypeResolver)(
```
Comment (Skylion007) [inline torch/csrc/jit/mobile/flatbuffer_loader.cpp:93]: Interesting use of typedef could "using" be used more cleanly here?

File: torch/csrc/jit/mobile/debug_info.cpp (hunk 2 above)
Comment (cyyever) [reply torch/csrc/jit/mobile/debug_info.cpp:161]: It takes const reference.

File: torch/csrc/jit/mobile/promoted_prim_ops.cpp (hunk 4 above)
Comment (cyyever) [reply torch/csrc/jit/mobile/promoted_prim_ops.cpp:98]: Applied

File: torch/csrc/jit/mobile/import.cpp (hunk 3 above)
Comment (cyyever) [reply torch/csrc/jit/mobile/import.cpp:148]: input is const IValue&, moving it has no effect.

File: torch/csrc/jit/mobile/debug_info.cpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/jit/mobile/debug_info.cpp:143]: val has type of const IValue&

File: torch/csrc/jit/mobile/flatbuffer_loader.cpp (hunk 5 above)
Comment (cyyever) [reply torch/csrc/jit/mobile/flatbuffer_loader.cpp]: Applied

File: torch/csrc/jit/mobile/debug_info.cpp (hunk 1 above)
Comment (Skylion007) [reply torch/csrc/jit/mobile/debug_info.cpp:143]: Oh, nvm then.

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/jit/mobile/flatbuffer_loader.cpp [hunk 6]
```diff
@@ -575,7 +575,7 @@ IValue parseTuple(
     const mobile::serialization::IValue& ivalue) {
   const auto& tuple = ivalue.val_as_Tuple();
   std::vector<IValue> res;
```
Comment (Skylion007) [inline torch/csrc/jit/mobile/flatbuffer_loader.cpp:577]: Nit: can you add a reserve here?

Comment (cyyever) [reply torch/csrc/jit/mobile/flatbuffer_loader.cpp:577]: Fixed

**Pull request closed without merging**